index.html

```diff
@@ -86,51 +112,240 @@
-      var yearRange = options.yearRange.split('-');
+    if (options.party !== 'all') {
+      var party = options.party;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.party === party;
       });
     }
-
+    
     // SCATTERPLOT
+    var points = chart.svg.selectAll('.point')
+      .data(txData);
 
+    points.enter().append('circle')
+      .classed("point", true)
+      .attr('r', 5);
+    
+    points
+      .classed('point',true)
+      .classed("R", function (d) {return d.party === "rep";})
+      .classed("D", function (d) {return d.party === "dem";})
+      .attr('cx', function (d){return chart.x(d.cfScore) ;})
+      .attr('cy', function (d){return chart.y(d.distPart) ;})
+      .on('mouseover', chart.tip.show)
+      .on('mouseout', chart.tip.hide);
+
+    points.exit().remove();
+    
     // HISTOGRAMS
+    // X-HIST
+ 
+    var dataHistX = d3.layout.histogram()
+      .bins(chart.x.ticks())
+      .value(function (d) {return d.cfScore} )
+      (txData);
+
+    var histy = d3.scale.linear()
+      .domain([0, d3.max(dataHistX, function (d){ return d.y; } ) ])
+      .range([ chart.y(chart.y.ticks()[0]) , chart.y(chart.y.ticks()[1]) ]);
+
+    var histx = d3.scale.linear()
+      .domain([0,11]) // needs to be updated to be dynamic
+      .range([0,width]);
+
+    var barWidthX = (histx(dataHistX[0].dx)+1)/2 ;
+
+    var barX = chart.svg.selectAll(".barX")
+      .data(dataHistX);
+
+    barX.enter()
+      .append("rect")
+      .attr("class", "barX")
+      .attr("x", function (d) { return chart.x(d.x) + barWidthX/2; })
+      .attr("width", barWidthX);
+
+    barX
+      .transition()
+      .attr("y", function (d) { return histy(d.y); })
+      .attr("height", function (d) { return height - margin.top - histy(d.y) ; })
+
+    // Y-HIST
+
+    var dataHistY = d3.layout.histogram()
+      .bins(chart.y.ticks())
+      .value(function (d) {return d.distPart} )
+      (txData);
+
+    var histx2 = d3.scale.linear()
+      .domain([0, d3.max(dataHistY, function (d){ return d.y; } ) ])
+      .range([ chart.x(chart.x.ticks()[0]) , chart.x(chart.x.ticks()[1])]);
+
+    var histy2 = d3.scale.linear()
+      .domain([0,4.5]) // needs to be updated to be dynamic
+      .range([0,height]);
+
+    var barWidthY = histy2(dataHistY[0].dx);
+
+    var barY = chart.svg.selectAll(".barY")
+      .data(dataHistY);
+
+    barY.enter()
+      .append("rect")
+      .attr("class", "barY")
+      .attr("y", function (d) { return chart.y(d.x) - (.75*barWidthY); })
+      .attr("height", function (d) { return barWidthY/2});
+
+    barY
+      .transition()
+      .attr("x", margin.left)
+      .attr("width", function (d) {return histx2(d.y)-margin.left });
+
+    // // LINE
+    var meanData = txData.map(function (d){return d.cfScore});
+
+    var meanVal = ss.mean(meanData);
+
+    var line = chart.svg.selectAll(".line")
+      .data(meanData);
```

Comment:
Why use meanData here? This is creating a line for each row of data, but we only need one.

You could use `.data([meanVal])`, though, and on the upside, you'd be able to use `d` when setting the `x1` and `x2` attributes. Finally, you wouldn't need to `.exit()`, because there'd only ever be a single data point.